From 55677450b3ae542b56df8df2267025692bdda200 Mon Sep 17 00:00:00 2001
From: Anthony Rabbito <hello@anthonyrabbito.com>
Date: Tue, 9 Jul 2024 11:49:33 -0400
Subject: [PATCH] ctx.patch

---
 src/main.cpp | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/src/main.cpp b/src/main.cpp
index e8ebd66..d975896 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -762,17 +762,13 @@ int main(int argc, char **argv)
 		}
 	}

-	if ( gamescope::Process::HasCapSysNice() )
 	{
 		gamescope::Process::SetNice( -20 );
+		fprintf(stdout, "WARNING: This version of gamescope is patched to assume a patched kernel to not require CAP_SYS_NICE for priority queues\n");

 		if ( g_bRt )
 			gamescope::Process::SetRealtime();
 	}
-	else
-	{
-		fprintf( stderr, "No CAP_SYS_NICE, falling back to regular-priority compute and threads.\nPerformance will be affected.\n" );
-	}

 #if 0
 	while( !IsInDebugSession() )
--
2.45.1
